src/main/java/org/folio/rest/impl/InitAPIs.java

```diff
@@ -78,11 +67,38 @@ public void init(Vertx vertx, Context context, Handler<AsyncResult<Boolean>> res
       });
   }
 
+  private void initDatabindCodec() {
+    var serializationConfig = ObjectMapperTool.getMapper().getSerializationConfig();
+    var deserializationConfig = ObjectMapperTool.getMapper().getDeserializationConfig();
+    DatabindCodec.mapper().setConfig(serializationConfig);
+    DatabindCodec.prettyMapper().setConfig(serializationConfig);
+    DatabindCodec.mapper().setConfig(deserializationConfig);
+    DatabindCodec.prettyMapper().setConfig(deserializationConfig);
+    log.info("initDatabindCodec:: Data bind codec was successfully started");
+  }
+
+  private void initSpringContext(Vertx vertx, Context context) {
+    SpringContextUtil.init(vertx, context, ApplicationConfig.class);
+    SpringContextUtil.autowireDependencies(this, context);
+    log.info("initSpringContext:: Spring context was successfully started");
+  }
+
+  private void initKafkaConsumersVerticles(Vertx vertx, Promise<Object> handler) {
+    deployKafkaConsumersVerticles(vertx).onComplete(ar -> {
+      if (ar.failed() && isConsumersVerticleMandatory) {
+        handler.fail(ar.cause());
+      } else {
+        handler.complete();
+      }
+    });
+  }
+
   private Future<?> deployKafkaConsumersVerticles(Vertx vertx) {
-    AbstractApplicationContext springContext = vertx.getOrCreateContext().get("springContext");
+    var springContext = (AbstractApplicationContext) vertx.getOrCreateContext().get(SPRING_CONTEXT_KEY);
     var consumers = List.of(
       deployVerticle("inventory-item-consumers", itemConsumersVerticleNumber, InventoryItemConsumersVerticle.class, vertx, springContext),
-      deployVerticle("inventory-holding-consumers", holdingConsumersVerticleNumber, InventoryHoldingConsumersVerticle.class, vertx, springContext),
+      deployVerticle("inventory-holding-create-consumers", holdingCreateConsumersVerticleNumber, InventoryHoldingCreateConsumersVerticle.class, vertx, springContext),
+      deployVerticle("inventory-holding-update-consumers", holdingUpdateConsumersVerticleNumber, InventoryHoldingUpdateConsumersVerticle.class, vertx, springContext),
```

Comment:
a bit worry about introducing new verticle. So we have worker verticle that will not be mixed with main event bus. But new worker thread has by default 20 threads in the pool as I remember(it is configurable). It can be so many for all kafka topics.
Initially thought that it would be good to have 1 verticle per item CRUD and 1 verticle per holding CRUD

Reply:
> a bit worry about introducing new verticle. So we have worker verticle that will not be mixed with main event bus. But new worker thread has by default 20 threads in the pool as I remember(it is configurable). It can be so many for all kafka topics. Initially thought that it would be good to have 1 verticle per item CRUD and 1 verticle per holding CRUD

How about reducing the number of worker threads?

Reply:
solved by adding configurable worker pool size